### DISCUSSION TITLE
Can GarnetServer.exe run as services?

### DISCUSSION THREAD

**Discussion by KaiJie999** [microsoft/garnet#301] [Q&A] [answered]
Can GarnetServer.exe run as services? Like SqlServer services or redis-server

**Comment by whippet-rider** [comment, marked as answer]
It appears so => https://microsoft.github.io/garnet/docs/getting-started/configuration

**Reply by christiantansastro** [reply]
Hi, I'm new to Garnet, could you help explain to me the steps on how to add GarnetServer.exe as a windows service? I've gone through the link you mentioned, but I still couldn't find a way to add it as a service.

**Reply by whippet-rider** [reply]
Actually to be honest, it looks like you would have to wrap garnet within a [windows service](https://learn.microsoft.com/en-us/dotnet/core/extensions/windows-service) project, like they have done via the [getting started](https://microsoft.github.io/garnet/docs/getting-started) details and creating a [GarnetServer](https://github.com/microsoft/garnet/blob/main/main/GarnetServer/Program.cs) application to allow installing as a windows service at least?

**Reply by whippet-rider** [reply]
Just quickly googled this example github project => https://github.com/MichalMoudry/garnet-as-win-service

**Reply by nightroman** [reply]
I use [nssm](https://github.com/kirillkovalenko/nssm) for low ceremony service management and PowerShell tools
https://github.com/nightroman/FarNet.Redis/tree/main/src/GarnetServer

**Comment by wessisaac** [comment]
Garnet as a service In Windows OS, 
1. download 64 bit package from Git-Hub release (https://github.com/microsoft/garnet/releases)
2. Place the downloaded package in your convenient place. Eg: C:\Program Files\Garnet
3. In windows, Garnet as a service is best. To install garnet as a service, download 64 bit nssm from the url "https://nssm.cc/download" and place the nssm.exe folder into your Garnet package folder
4. In the command prompt enter the command, >nssm install Garnet
5. A GUI will open, in the path text box, you can browse your GarnetServer.exe path
      Path: 	C:\Program Files\Garnet\GarnetServer.exe
6. In the argument text box, please enter the below command with path of your Garnet configuration file
     Arguments:	GarnetServer.exe --config-import-path "C:\Program Files\Garnet\Garnet.conf"
7. Then click install button
Now you can access Garnet in windows 

Use Redis-cli in windows:
To use redis-cli in windows, download redis-windows package from the path "https://github.com/zkteco-home/redis-windows/releases/tag/7.4.2", then copy and paste the redis-cli.exe file (redis-clie.exe single file is enough) in Garnet package folder.
In command prompt, you can place the Garnet path, then redis-cli commands can be used.
